python/cuml/test/test_fil.py

```diff
@@ -130,14 +131,66 @@ def test_fil_classification(n_rows, n_columns, num_rounds, tmp_path):
     assert array_equal(fil_preds, xgb_preds_int)
 
 
-@pytest.mark.parametrize('n_rows', [10000])
+@pytest.mark.parametrize('n_rows', [unit_param(1000), quality_param(10000),
+                         stress_param(500000)])
+@pytest.mark.parametrize('n_columns', [unit_param(20), quality_param(100),
+                         stress_param(1000)])
+@pytest.mark.parametrize('num_rounds', [unit_param(5), quality_param(10),
+                         stress_param(90)])
+@pytest.mark.parametrize('max_depth', [unit_param(3),
+                                       unit_param(7),
+                                       stress_param(11)])
+@pytest.mark.skipif(has_xgboost() is False, reason="need to install xgboost")
+def test_fil_regression(n_rows, n_columns, num_rounds, tmp_path, max_depth):
+    # settings
+    classification = False  # change this to false to use regression
+    n_rows = n_rows  # we'll use 1 millions rows
+    n_columns = n_columns
+    random_state = np.random.RandomState(43210)
+
+    X, y = simulate_data(n_rows, n_columns,
+                         random_state=random_state,
+                         classification=classification)
+    # identify shape and indices
+    n_rows, n_columns = X.shape
+    train_size = 0.80
+
+    X_train, X_validation, y_train, y_validation = train_test_split(
+        X, y, train_size=train_size, random_state=0)
+
+    model_path = os.path.join(tmp_path, 'xgb_reg.model')
+    bst = _build_and_save_xgboost(model_path, X_train,
+                                  y_train,
+                                  num_rounds,
+                                  classification,
+                                  xgboost_params={'max_depth': max_depth})
+
+    dvalidation = xgb.DMatrix(X_validation, label=y_validation)
+    xgb_preds = bst.predict(dvalidation)
+
+    xgb_mse = mean_squared_error(y_validation, xgb_preds)
+    print("Reading the saved xgb model")
+    fm = ForestInference.load(model_path,
+                              algo='BATCH_TREE_REORG',
+                              output_class=False)
+    fil_preds = np.asarray(fm.predict(X_validation))
+    fil_mse = mean_squared_error(y_validation, fil_preds)
+
+    print("XGB accuracy = ", xgb_mse, " Forest accuracy: ", fil_mse)
+    assert fil_mse == pytest.approx(xgb_mse, 0.01)
```

Comment:
probably just the array_equal assert below is sufficient

Reply:
The xgboost test is not part of this pull request.

I'll remove the `print()` calls, but otherwise will leave it as is.